lib/posix/pthread.c

```diff
@@ -299,6 +303,28 @@ int pthread_getschedparam(pthread_t pthread, int *policy,
 	return 0;
 }
 
+/**
+ * @brief Dynamic package initialization
+ *
+ * See IEEE 1003.1
+ */
+int pthread_once(pthread_once_t *once, void (*init_func)(void))
+{
+	pthread_mutex_lock(&pthread_key_lock);
```

Comment:
Why use a pthread lock here instead of a semaphore?  You're using a k_sem below for a lock...

Reply:
No particular reason other than that most `pthread` functions in the pthread.c file used `pthread_mutex_lock` functions (like pthread_cancel, pthread_create, etc.). So in keeping with that, I used this for pthread_once and the addition in pthread_exit